app/Actions/DataFixes/ReorderStreamMessages.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace App\Actions\DataFixes;
+
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Log;
+use Lorisleiva\Actions\Concerns\AsCommand;
+
+class ReorderStreamMessages
+{
+    use AsCommand;
+
+    public string $commandSignature = 'data-fix:reorder-stream-messages';
+
+    public function handle()
+    {
+        $this->backupStreamMessages();
+        $this->createTempTable();
+        $this->loadTempTable();
+        $this->truncateStreamMessages();
+        
+        try {
+            $this->reloadStreamMessages();
+        } catch (\Exception $e) {
+            $this->restoreStreamMessages();
+            Log::error('Error reloading stream messages. Original stream_message records restored. Error: ' . $e->getMessage());
+        }
+        
+        $this->cleanup();
+
+    }
+
+    /**
+     * Backup the stream messages table in case something goes wrong
+     */
+    private function backupStreamMessages(): void
+    {
+        $sql = 'CREATE TABLE IF NOT EXISTS `sm_backup` AS SELECT * FROM `stream_messages`';
+
+        DB::statement($sql);
+    }
+    
+    /**
+     * Create a temp table to hold the ordered stream messages
+     */
+    private function createTempTable()
+    {
+        $sql = <<<SQL
+            CREATE TEMPORARY TABLE IF NOT EXISTS sm_temp (
+                id int auto_increment Primary key, 
+                topic varchar(127) not null,
+                message json,
+                sent_at datetime,
+                error TEXT,
+                created_at datetime,
+                updated_at datetime,
+                orig_id int
+            )
+        SQL;
+
+        DB::statement($sql);
+    }
+
+    private function loadTempTable(): void
+    {
+        $sql = <<<SQL
+            INSERT INTO sm_temp (topic, `message`, sent_at, error, created_at, updated_at, orig_id)
+            SELECT topic, `message`, NULL, error, created_at, updated_at, id AS orig_id FROM stream_messages 
+            order by created_at
+        SQL;
+
+        $results = DB::statement($sql);
+
+    }
+    
+    
+    /**
+     * Truncate the stream messages table
+     */
+    private function truncateStreamMessages(): void
+    {
+        DB::table('stream_messages')->truncate();
+    }
+
+    /**
+     * Insert ordered stream messages from temp table
+     **/
+    private function reloadStreamMessages(): void
+    {
+        $sql = <<<SQL
+            INSERT INTO stream_messages (id, topic, `message`, sent_at, error, created_at, updated_at)
+            SELECT id, topic, `message`, sent_at, error, created_at, updated_at FROM sm_temp
+        SQL;
+
+        DB::statement($sql);
+    }
+
+    /**
+     * Restore the stream messages table from the backup table
+     */
+    private function restoreStreamMessages(): void
```

Comment:
Hopefully only need to ever run this once and hopefully it will work, but I think there might be some weird behavior if `reloadStreamMessages` starts but has an error partway through-- in that case, this would just add the backed-up stream messages AFTER the partially-loaded re-written ones.

Unfortunately, the TRUNCATE done earlier in the process gives an implicit COMMIT (so we cannot just wrap the whole thing in a transaction), but I think it would clear this case up if the code re-TRUNCATED before trying to restore the backed-up messages if the earlier step failed.